airflow/api/common/experimental/cancel_dag_run.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""Cancel DAG run API."""
+from typing import Optional, List, Dict, Any
+
+from flask import url_for
+
+from airflow.api.common.experimental import check_and_get_dag
+from airflow.exceptions import AirflowBadRequest
+from airflow.models import DagRun, cancel_task_instances
+from airflow.utils.db import provide_session
+from airflow.utils.state import State
+
+
+@provide_session
+def cancel_dag_run(dag_id, dagrun_run_id, session=None):
```

Comment:
missing mypy type of session

Reply:
If Airflow doesn't have mypy types then ii guess remove the typing below. it's wrong anyway...

Reply:
Updated.